Source/ASDisplayNode+Layout.mm

```diff
@@ -883,6 +887,28 @@ - (void)_completePendingLayoutTransition
   }
 }
 
+/**
+ * In cases where a late layout tree flatten is performed, this is used to perform the transition copy and flatten.
+ * This is import to support const pointer usage instead of shared_ptr when managing layouts and transitions
+ * during the layout flatten/ transition lifecycle.
+ */
+- (ASLayoutTransition *)_locked_flattenLayoutTransitionIfNeeded:(ASLayoutTransition *)pendingLayoutTransition {
+  ASLayoutTransition *transition = pendingLayoutTransition;
+  if ([ASDisplayNode shouldStoreUnflattenedLayouts]) {
+    ASDisplayNodeLayout pendingLayout = pendingLayoutTransition.pendingLayout;
+    _unflattenedLayout = pendingLayout.layout;
```

Comment:
The use case for this method isn't 100% perfect [here](https://github.com/TextureGroup/Texture/pull/1156/files#diff-71ea742a29543cb7e08c3dffe2a2508eR667), so it seems like it's only once [here](https://github.com/TextureGroup/Texture/pull/1156/files#diff-71ea742a29543cb7e08c3dffe2a2508eR879). What do you think about doing everything here inline?

Reply:
Yes that sounds like a good idea, I would prefer to group everything together for clarity but there are no truly common cases.